sdk/storage/azblob/README.md

```diff
@@ -1,443 +1,276 @@
 
-cred, err := service.NewSharedKeyCredential(accountName, accountKey)
-handleError(err)
+```go
+// create a credential for authenticating with Azure Active Directory
+cred, err := azidentity.NewDefaultAzureCredential(nil)
+// TODO: handle err
 
-serviceClient, err := service.NewClientWithSharedKey(serviceURL, cred, nil)
-handleError(err)
-
-fmt.Println(serviceClient.URL())
+// create an azblob.Client for the specified storage account that uses the above credential
+client, err := azblob.NewClient("https://MYSTORAGEACCOUNT.blob.core.windows.net/", cred, nil)
+// TODO: handle err
 ```
 
-For more information about blob service client and how to configure custom domain names for Azure Storage check out
-the [official documentation][azure_portal_account_url]
-
-#### Types of credentials
+Learn more about enabling Azure Active Directory for authentication with Azure Storage in [our documentation][storage_ad] and [our samples](#next-steps).
 
-The azblob clients support authentication via Shared Key Credential, Connection String, Shared Access Signature, or any
-of the `azidentity` types that implement the `azcore.TokenCredential` interface.
+## Key concepts
 
-##### 1. Creating the client from a shared key
+Blob storage is designed for:
 
-To use an account [shared key][azure_shared_key] (aka account key or access key), provide the key as a string. This can
-be found in your storage account in the [Azure Portal][azure_portal_account_url] under the "Access Keys" section or by
-running the following Azure CLI command:
+- Serving images or documents directly to a browser.
+- Storing files for distributed access.
+- Streaming video and audio.
+- Writing to log files.
+- Storing data for backup and restore, disaster recovery, and archiving.
+- Storing data for analysis by an on-premises or Azure-hosted service.
 
-```bash
-az storage account keys list -g my-resource-group -n mystorageaccount
-```
+Blob storage offers three types of resources:
 
-Use Shared Key authentication as the credential parameter to authenticate the client:
+- The _storage account_
+- One or more _containers_ in a storage account
+- One ore more _blobs_ in a container
 
-```golang
-import "github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/service"
+Instances of the `azblob.Client` type provide methods for manipulating containers and blobs within a storage account.
+The storage account is specified when the `azblob.Client` is constructed.
+Use the appropriate client constructor function for the authentication mechanism you wish to use.
 
-serviceURL := fmt.Sprintf("https://%s.blob.core.windows.net/", accountName)
+Learn more about options for authentication _(including Connection Strings, Shared Key, Shared Key Signatures, Active Directory, and anonymous public access)_ [in our examples.](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/storage/azblob/examples_test.go)
```

Comment:
Minor: I didn't find those terms in the examples file, so it's hard to find the right thing if I got there from this paragraph.

Might be worth adding some comments so they're ctrl-f'able.

Reply:
Good idea!

Reply:
I added the appropriate comments and one new example.  I don't have one for SAS at present though, to be added later.